### PR TITLE
Fix win static build

### PR DESCRIPTION
**Pull request by SweetVishnya**
ref #1070

### CODE REVIEW

**Review by mrexodia — CHANGES_REQUESTED** [review]

File: CMakeModules/FindBITWUZLA.cmake [hunk 1]
```diff
@@ -21,6 +21,11 @@ if(NOT BITWUZLA_LIBRARIES)
     set(BITWUZLA_LIBRARIES "$ENV{BITWUZLA_LIBRARIES}")
 endif()
 
+if(BITWUZLA_LIBRARIES)
+    # Link with pthread
+    set(BITWUZLA_LIBRARIES ${BITWUZLA_LIBRARIES} pthread)
```
[the remark this answers is not in the document]
Comment (SweetVishnya) [reply CMakeModules/FindBITWUZLA.cmake]: Removed pthread, waiting for our local CI to pass. `pthread` should be better linked by an end client writing a Triton tool.

File: src/libtriton/CMakeLists.txt [hunk 2]
```diff
@@ -244,6 +244,11 @@ if(NOT MSVC AND (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID
     target_compile_options(triton PRIVATE -std=c++11)
 endif()
 
+# Enable static msvc runtime.
+if(MSVC AND MSVC_STATIC)
+    target_compile_options(triton PRIVATE "$<IF:$<CONFIG:Debug>,/MTd,/MT>")
```
[the remark this answers is not in the document]
Comment (SweetVishnya) [reply src/libtriton/CMakeLists.txt:249]: kk, will replace to target property

File: CMakeModules/FindBITWUZLA.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrexodia) [reply CMakeModules/FindBITWUZLA.cmake]: Yeah, I forgot to submit my review yesterday 🤦‍♂️

If pthread is a dependency of Triton it should just be in here though and also in the package config file.

[the remark this answers is not in the document]
Comment (SweetVishnya) [reply CMakeModules/FindBITWUZLA.cmake]: pthread is a platform specific dependency of Bitwuzla only

File: src/libtriton/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (SweetVishnya) [reply src/libtriton/CMakeLists.txt]: property does not work(((

[the remark this answers is not in the document]
Comment (SweetVishnya) [reply src/libtriton/CMakeLists.txt]: Something in CMake still inserts `-MD`, so, I'll revert the changes

**Comment by mrexodia** [comment]
Are you using CMake > 3.15? This policy should be set to NEW (should default to new with 3.15): https://cmake.org/cmake/help/latest/policy/CMP0091.html#policy:CMP0091

**Comment by SweetVishnya** [comment]
> Are you using CMake > 3.15? This policy should be set to NEW (should default to new with 3.15): https://cmake.org/cmake/help/latest/policy/CMP0091.html#policy:CMP0091

I am using CMake 3.20. In my experience cmake does not always work the same as it says. So, I prefer a working solution instead of canonical one. Moreover, I do not see much difference between `/MT` and `MultiThreaded`.

**Comment by SweetVishnya** [comment]
Anyway, I'll give it one more try.

**Comment by SweetVishnya** [comment]
My bad, it was a stupid typo.

**Comment by mrexodia** [comment]
🚀

**Comment by SweetVishnya** [comment]
@JonathanSalwan, all our internal CI passes. We can finally merge everything (including this PR) to master.

**Pull request merged**